store/file/ods_q4.go

```diff
@@ -60,41 +61,36 @@ func CreateODSQ4(
 	return nil
 }
 
-// OpenODSQ4 opens ODS file under the given FS path. The Q4 is opened lazily
-// on demand.
-func OpenODSQ4(pathODS, pathQ4 string) (*ODSQ4, error) {
-	ods, err := OpenODS(pathODS)
-	if err != nil {
-		return nil, fmt.Errorf("failed to open ODS: %w", err)
-	}
-
+// ODSWithQ4 returns ODSQ4 instance for the given FS paths. It opens Q4 file lazily.
+func ODSWithQ4(ods *ODS, pathQ4 string) *ODSQ4 {
 	return &ODSQ4{
-		ods: ods,
-		q4Open: func() (*Q4, error) {
-			return OpenQ4(pathQ4)
-		},
-	}, nil
+		ods:    ods,
+		pathQ4: pathQ4,
+	}
 }
 
-func (odsq4 *ODSQ4) getQ4() (eds.Accessor, error) {
+func (odsq4 *ODSQ4) tryLoadQ4() {
 	if odsq4.q4Opened.Load() {
-		return odsq4.q4, nil
+		return
 	}
 
 	odsq4.q4Mu.Lock()
 	defer odsq4.q4Mu.Unlock()
-	if odsq4.q4Opened.Load() {
-		return odsq4.q4, nil
+	if loaded := odsq4.q4Opened.Swap(true); loaded {
```

Comment:
Imo, this needs a note anyway on why its Swap and then we can avoid creating loaded